### PR TITLE
feat: add support for ha registry

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: pkg/addons/applier.go [hunk 1]
```diff
@@ -150,19 +151,13 @@ func (a *Applier) GenerateHelmConfigsForRestore() ([]v1beta1.Chart, []v1beta1.Re
 	return charts, repositories, nil
 }
 
+// GetAirgapCharts is deprecated in favor of GetBuiltinCharts.
 func (a *Applier) GetAirgapCharts() ([]v1beta1.Chart, []v1beta1.Repository, error) {
```
Comment (laverya) [inline pkg/addons/applier.go:155]: you can totally remove this function if you want
its only use should be in the version metadata command, no?

File: pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go [hunk 2]
```diff
@@ -66,6 +66,7 @@ func init() {
 		helmValues["image"] = map[string]interface{}{
 			"repository": parts[0],
 			"tag":        parts[1],
+			"pullPolicy": "Always",
```
Comment (laverya) [inline pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go:69]: please don't actually commit this

File: Makefile [hunk 3]
```diff
@@ -11,11 +11,14 @@ EMBEDDED_OPERATOR_CHART_URL = oci://registry.replicated.com/library
 EMBEDDED_OPERATOR_CHART_NAME = embedded-cluster-operator
 EMBEDDED_OPERATOR_CHART_VERSION = 0.32.2
 EMBEDDED_OPERATOR_UTILS_IMAGE = busybox:1.36.1
-EMBEDDED_CLUSTER_OPERATOR_IMAGE_OVERRIDE =
+EMBEDDED_CLUSTER_OPERATOR_IMAGE_OVERRIDE = ttl.sh/ethan/embedded-cluster-operator-image:24h
```
Comment (emosbaugh) [inline Makefile:14]: DONT MERGE THIS

File: pkg/addons/seaweedfs/seaweedfs.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	if err := yaml.Unmarshal(helmValuesYAML, &helmValues); err != nil {
+		panic(fmt.Errorf("failed to unmarshal helm values: %w", err))
+	}
+	initExistingConfigSecret()
+}
+
+func initExistingConfigSecret() {
+	if val, ok := helmValues["filer"].(map[interface{}]interface{}); ok {
+		if val, ok := val["s3"].(map[interface{}]interface{}); ok {
+			val["existingConfigSecret"] = s3SecretName
```
Comment (sgalsaleh) [inline pkg/addons/seaweedfs/seaweedfs.go:112]: I see the secret name is already hardcoded in the helm values YAML file. Any reason it needs to be initialized here too?

File: .github/actions/e2e/action.yml [hunk 5]
```diff
@@ -131,6 +131,6 @@ runs:
       echo -e "\e[1;33m                                                                                                        \e[0m"
       echo -e "\e[1;33m└──────────────────────────────────────────────────────────────────────────────────────────────────────┘\e[0m"
     shell: bash
-  # - name: Setup upterm session (ssh)
-  #   uses: lhotari/action-upterm@v1
-  #   if: always()
+  - name: Setup upterm session (ssh)
```
Comment (emosbaugh) [inline .github/actions/e2e/action.yml:134]: TODO: remove this before merge

**Pull request closed without merging**